sw/device/tests/spi_device_ottf_console_test.c

```diff
@@ -0,0 +1,24 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_spi_device.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG(.console.type = kOttfConsoleSpiDevice,
+                        .console.base_addr = TOP_EARLGREY_SPI_DEVICE_BASE_ADDR,
+                        .console.test_may_clobber = false, );
+
+static const char test_str[] = "This is a SPI device OTTF console test.";
+
+bool test_main(void) {
+  for (int i = 0; i < 100; i++) {
+    LOG_INFO("%s round: %d", test_str, i);
```

Comment:
maybe this test should try writing varying size strings to the console? rather than the same length string each time? 

Also add some intentional corner cases, e.g.:
- string bigger than the spi flash egress buffer
- empty string
- string that wraps around buffer but is smaller than egress buffer, etc.